File: Algorithms/Easy/467_next_lucky_ticket/solution.dart
// Solution for https://coderun.yandex.ru/problem/next-lucky-ticket
// Other solutions: https://github.com/Melodiz/CodeRun

import 'dart:io';

bool isLucky(int k) {
  String numStr = k.toString();
  int firstSum = 0;
  int secondSum = 0;
  
  for (int i = 0; i < 3; i++) {
    firstSum += int.parse(numStr[i]);
    secondSum += int.parse(numStr[i + 3]);
  }
  
  return firstSum == secondSum;
}

void main() {
  int n = int.parse(stdin.readLineSync()!);
  
  for (int k = n + 1; k < 1000000; k++) {
    if (isLucky(k)) {
      print(k);
      break;
    }
  }
}

File: Algorithms/Easy/467_next_lucky_ticket/solution.py
# Solution for https://coderun.yandex.ru/problem/next-lucky-ticket
# Other solutions: https://github.com/Melodiz/CodeRun

def main():
    n = int(input())
    for k in range(int(n)+1, 1000000):
        if is_lucky(k):
            print(k)
            break
    return

def is_lucky(k):
    return sum(map(int, str(k)[:3])) == sum(map(int, str(k)[3:]))

if __name__ == "__main__":
    main()
